contracts/test/auth.ts

```diff
@@ -149,20 +154,14 @@ describe('Auth', function () {
     await expect(siweAuthTests.testVerySecretMessage(bearer3)).to.be.reverted;
 
     // Expired bearer.
-    const siweStr4 = await siweMsg('localhost', 0, new Date(Date.now() + 10)); // Expire after 0.01 seconds.
+    const expiration = new Date(Date.now() + 500);
+    const siweStr4 = await siweMsg('localhost', 0, expiration); // Expire after 0.5 seconds.
     const bearer4 = await siweAuthTests.testLogin(
       siweStr4,
       await erc191sign(siweStr4, account),
     );
-    // Wait for 2 blocks. The first block may still have a timestamp earlier than the expiration time.
-    await new Promise(async (r) => {
-      const desiredBlockNumber = (await ethers.provider.getBlockNumber()) + 2;
-      ethers.provider.on('block', (blockNumber) => {
-        if (blockNumber == desiredBlockNumber) {
-          r();
-        }
-      });
-    });
+    await delay(expiration.getTime() - Date.now());
+    await siweAuthTests.doNothing();
```

Comment:
We need this call? 🤔

Reply:
Block times aren't metronomic and the time experienced in view calls is that of whatever the last mined block was.

So I'm waiting until I 100% know that it will be expired, then I push a transaction to force a block to be mined and to wait for the next block, then the view call executes in the context of the last block whos timestamp will always be after the bearer token expiration.

Reply:
I think ` await siweAuthTests.doNothing();` doesn't assure you that the block will be mined, just that the tx is submitted? You need to wait for the receipt.

Reply:
Good catch, somehow that slipped my mind.